app/discovery/provider/consulcatalog/client.go

```diff
@@ -0,0 +1,147 @@
+package consulcatalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Client struct {
+	address string
+	client  *http.Client
+}
+
+func NewClient(address string, timeout time.Duration) *Client {
+	cl := &Client{
+		address: strings.TrimRight(address, "/"),
+	}
+
+	cl.client = &http.Client{
+		Timeout: timeout,
+	}
+
+	return cl
+}
+
+func (cl *Client) Get() ([]*ConsulService, error) {
+	var result []*ConsulService
+
+	serviceNames, err := cl.getServiceNames()
+	if err != nil {
+		return nil, fmt.Errorf("error get service names, %w", err)
+	}
+
+	for _, serviceName := range serviceNames {
```

Comment:
this is confusing. It gets services names first and for service name get services? If those consuing naming is a part of consul we need a good comment explaing what is going on here and why